Optimiser.fs

```diff
@@ -481,6 +493,15 @@ module Graph =
                                               | _ -> false))
                      ps)
 
+    /// Decides whether a given Command contains any `assume` command
+    /// in any of the sequentially composed primitives inside it
+    let hasAssume : Command -> bool =
+        fun c ->
```

Comment:
I've tried to keep a style

```
name : A -> B -> C =
    fun a b -> c
```

As I think it makes it clearer (specification then implementation), and fits the very Haskell-like codebase and use of types better ( i think ).